src/transformStateWithClones.js

```diff
@@ -6,8 +6,51 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  // const newObject = {};
+
+  const clone = {}; // новий порожній об'єкт
+  const resultArray = [];
+
+  for (const key in state) {
+    clone[key] = state[key];
+  }
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(clone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(clone, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clearProperties(clone);
+        break;
+    }
+    resultArray.push({ ...clone });
+  }
+
+  return resultArray;
+}
+
+function addProperties(clone, extraData) {
+  Object.assign(clone, extraData);
+}
```

Comment:
The function `addProperties` is correctly using `Object.assign` to add properties to the clone. However, the parameter name `clone` should be renamed to `stateCopy` to reflect the naming convention.